pkg/ingester/limiter.go

```diff
@@ -210,34 +211,55 @@ func (l *Limiter) convertGlobalToLocalLimit(userID string, globalLimit int) int
 		return 0
 	}
 
-	// Given we don't need a super accurate count (ie. when the ingesters
-	// topology changes) and we prefer to always be in favor of the tenant,
-	// we can use a per-ingester limit equal to:
-	// (global limit / number of ingesters) * replication factor
-	numIngesters := l.ring.HealthyInstancesCount()
+	healthyIngestersCount := l.ring.HealthyInstancesCount()
 
 	// May happen because the number of ingesters is asynchronously updated.
 	// If happens, we just temporarily ignore the global limit.
-	if numIngesters == 0 {
+	if healthyIngestersCount == 0 {
 		return 0
 	}
```

Comment:
I think you can remove this whole code block. You already have `if ingestersInZoneCount == 0 {` below.